R/generateEDGEdata.R

```diff
@@ -384,10 +383,10 @@ toolGenerateEDGEdata <- function(input_folder, output_folder, cache_folder = NUL
       reg_demreg_tab <- fread(regional_demreg.path, header = TRUE)
 
       demscen_factors <- NULL
-      if (!is.null(demScen)) {
-        demscen.path <- system.file("extdata", "demscen_factors.csv", package="edgeTransport")
-        demscen_factors <- fread(demscen.path, header = TRUE)[demScen == demandScen]
-      }
+
+      demscen.path <- system.file("extdata", "demscen_factors.csv", package="edgeTransport")
+      demscen_factors <- fread(demscen.path, header = TRUE)[demandScen == demScen]
```

Comment:
so this is essentially an empty data.table for the `default` scenario. So this should work with the join which happens above after the regression (did you test it?).

Reply:
Yes I tested it - the joind happens only if nrow > zero